src/devices/MemoryLCD/LSxxxB7DHxx.cs

```diff
@@ -0,0 +1,256 @@
+        /// </summary>
+        private readonly int _extcomin;
+        #endregion
+
+        private readonly bool _shouldDispose;
+        private GpioController? _gpio;
+
+        private SpiDevice _spi;
+
+        internal byte[] _lineNumberBuffer;
+        internal byte[] _frameBuffer;
+
+        /// <summary>
+        /// Create a memory LCD device
+        /// </summary>
+        /// <param name="spi">SPI controller<br/><b>ChipSelectLineActiveState</b> must be <b>HIGH</b> or use <paramref name="gpio"/> and <paramref name="chipSelect"/>. See Datasheet 6-3</param>
+        /// <param name="gpio">GPIO controller</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        /// <param name="chipSelect">Chip select signal<br/>-1 when using SPI chipSelect line</param>
+        /// <param name="display">Display ON/OFF signal</param>
+        /// <param name="externalCom">External COM inversion signal input</param>
+        internal LSxxxB7DHxx(SpiDevice spi, GpioController? gpio = null, bool shouldDispose = true, int chipSelect = -1, int display = -1, int externalCom = -1)
+        {
+            _spi = spi ?? throw new ArgumentNullException(nameof(spi));
+
+            if (chipSelect != -1 || display != -1 || externalCom != -1)
+            {
+                _shouldDispose = gpio == null || shouldDispose;
+                _gpio = gpio;
+            }
+            else
+            {
+                _shouldDispose = false;
+                _gpio = null;
+            }
+
+            _scs = chipSelect;
+            _disp = display;
+            _extcomin = externalCom;
+
+            BytesPerLine = (PixelWidth + 7) / 8;
+
+            _lineNumberBuffer = Enumerable.Range(0, PixelHeight).Select(m => (byte)m).ToArray();
+            _frameBuffer = new byte[BytesPerLine * PixelHeight];
+
+            Init();
+        }
+
+        /// <summary>
+        /// Updates data of only one specified line.
+        /// </summary>
+        public void DataUpdate1Line(byte lineIndex, Span<byte> data, bool frameInversion = false)
+        {
+            if (data.Length != BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {PixelWidth}ck ({BytesPerLine} bytes)");
+            }
+
+            // m(1), ag(1), d(18), dummy(2)
+            Span<byte> outputBuffer = stackalloc byte[BytesPerLine + 4];
+
+            outputBuffer[0] = (byte)(ModeSelectionPeriodByte.Mode | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+            outputBuffer[1] = AgBytes.Table[lineIndex];
+            data.CopyTo(outputBuffer.Slice(2, BytesPerLine));
+
+            WriteSpi(outputBuffer);
+        }
+
+        /// <summary>
+        /// Updates arbitrary multiple lines data.
+        /// </summary>
+        public void DataUpdateMultipleLines(Span<byte> lineIndex, Span<byte> data, bool frameInversion = false)
```

Comment:
```suggestion
        public void UpdateMultipleRows(Span<int> rowIndex, Span<byte> data, bool frameInversion = false)
```
Similar to the other comment about updating the name.